src/tiledbsoma_ml/pytorch.py

```diff
@@ -318,25 +314,31 @@ def __len__(self) -> int:
 
     @property
     def shape(self) -> Tuple[int, int]:
-        """Get the approximate shape of the data that will be returned by this :class:`tiledbsoma_ml.ExperimentAxisQueryIterable`.
-        This is the number of obs (cell) and var (feature) counts in the returned data. If used in multiprocessing mode
-        (i.e. :class:`torch.utils.data.DataLoader` instantiated with num_workers > 0), the obs (cell) count will reflect
-        the size of the data partition assigned to the active process.
+        """Return the number of batches and features that will be yielded from this :class:`tiledbsoma_ml.ExperimentAxisQueryIterable`.
+
+        If used in multiprocessing mode (i.e. :class:`torch.utils.data.DataLoader` instantiated with num_workers > 0),
+        the number of batches will reflect the size of the data partition assigned to the active process.
 
         Returns:
-            A tuple of two ``int`` values: number of obs, number of vars.
+            A tuple of two ``int`` values: number of batches, number of vars.
 
         Lifecycle:
             experimental
         """
         self._init_once()
         assert self._obs_joinids is not None
         assert self._var_joinids is not None
-        world_size, _ = _get_distributed_world_rank()
-        n_workers, _ = _get_worker_world_rank()
-        partition_len = len(self._obs_joinids) // world_size // n_workers
-        div, rem = divmod(partition_len, self.batch_size)
-        return div + bool(rem), len(self._var_joinids)
+        world_size, rank = _get_distributed_world_rank()
+        n_workers, worker_id = _get_worker_world_rank()
+        obs_per_proc, obs_rem = divmod(len(self._obs_joinids), world_size)
+        # obs rows assigned to this "distributed" process
+        n_proc_obs = obs_per_proc + bool(rank < obs_rem)
```

Comment:
I believe this is incorrect.  Every GPU gets the same number of samples (this is a hard requirement).  Counts can vary across multiple data loader workers, but each GPU worker must have exactly the same sample count.  See notes in `_create_obs_joinids_partition`.

If the partitioning across GPUs does not have the same carddinatlity, you get a crash or stall when using DDP.

Reply:
Thanks for explaining, I've updated it to reflect that "distributed" processes get rounded-down splits, but their child "data-loader" processes can have ±0.5.

I'll add some simple tests of this math as well, later. I think it's worth codifying our assumptions/intentions here.